pkg/cmd/gpg-key/list/list.go

```diff
@@ -0,0 +1,108 @@
+package list
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/utils"
+	"github.com/spf13/cobra"
+)
+
+type ListOptions struct {
+	IO         *iostreams.IOStreams
+	Config     func() (config.Config, error)
+	HTTPClient func() (*http.Client, error)
+}
+
+func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Command {
+	opts := &ListOptions{
+		IO:         f.IOStreams,
+		Config:     f.Config,
+		HTTPClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "Lists GPG keys in your GitHub account",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if runF != nil {
+				return runF(opts)
+			}
+			return listRun(opts)
+		},
+	}
+
+	return cmd
+}
+
+func listRun(opts *ListOptions) error {
+	apiClient, err := opts.HTTPClient()
+	if err != nil {
+		return err
+	}
+
+	cfg, err := opts.Config()
+	if err != nil {
+		return err
+	}
+
+	host, err := cfg.DefaultHost()
+	if err != nil {
+		return err
+	}
+
+	gpgKeys, err := userKeys(apiClient, host, "")
+	if err != nil {
+		if errors.Is(err, scopesError) {
+			cs := opts.IO.ColorScheme()
+			fmt.Fprint(opts.IO.ErrOut, "Error: insufficient OAuth scopes to list GPG keys\n")
+			fmt.Fprintf(opts.IO.ErrOut, "Run the following to grant scopes: %s\n", cs.Bold("gh auth refresh -s read:gpg_key"))
+			return cmdutil.SilentError
+		}
+		return err
+	}
+
+	if len(gpgKeys) == 0 {
+		fmt.Fprintln(opts.IO.ErrOut, "No GPG keys present in GitHub account.")
+		return cmdutil.SilentError
+	}
+
+	t := utils.NewTablePrinter(opts.IO)
+	cs := opts.IO.ColorScheme()
+	now := time.Now()
+
+	for _, gpgKey := range gpgKeys {
+		t.AddField(gpgKey.Emails.String(), nil, nil)
+
+		t.AddField(gpgKey.KeyId, nil, nil)
```

Comment:
Should the ID field come first?

Reply:
I'd say it probably should do.

Reply:
I disagree here, I think the email associated with the key is how most people will recognize which key is which if there are multiple associated with an account since they don't have names like SSH keys. Additionally the UI on the GitHub settings page, as well as other GPG keystores I have seen, usually list email first.